@garbear you have a build failure :)

```
08:16:39 /home/jenkins/workspace/LINUX-64/xbmc/games/windows/GUIWindowGames.cpp: In member function ‘virtual bool KODI::GAME::CGUIWindowGames::GetDirectory(const string&, CFileItemList&)’:
08:16:39 /home/jenkins/workspace/LINUX-64/xbmc/games/windows/GUIWindowGames.cpp:269:29: error: binding reference of type ‘CFileItemPtr& {aka std::shared_ptr<CFileItem>&}’ to ‘const std::shared_ptr<CFileItem>’ discards qualifiers
08:16:39    for (CFileItemPtr& item : items)
08:16:39                              ^~~~~
```

--- a/system/playercorefactory.xml
+++ b/system/playercorefactory.xml
@@ -26,7 +26,7 @@
 
     <!-- DVDs -->
     <rule name="dvd" dvd="true" player="VideoPlayer" />
-    <rule name="dvdimage" dvdimage="true" game="false" player="VideoPlayer" />
+    <rule name="discimage" discimage="true" game="false" player="VideoPlayer" />
 
     <!-- Only VideoPlayer can handle these normally -->
     <rule name="sdp/asf" filetypes="sdp|asf" player="VideoPlayer" />

--- a/xbmc/FileItem.h
+++ b/xbmc/FileItem.h
@@ -758,12 +758,12 @@ public:
 
   void ClearSortState();
 
-  VECFILEITEMS::const_iterator begin() { return m_items.cbegin(); }
-  VECFILEITEMS::const_iterator end() { return m_items.cend(); }
+  VECFILEITEMS::iterator begin() { return m_items.begin(); }
+  VECFILEITEMS::iterator end() { return m_items.end(); }
   VECFILEITEMS::const_iterator begin() const { return m_items.begin(); }
   VECFILEITEMS::const_iterator end() const { return m_items.end(); }
-  VECFILEITEMS::const_iterator cbegin() const { return m_items.begin(); }
-  VECFILEITEMS::const_iterator cend() const { return m_items.end(); }
+  VECFILEITEMS::const_iterator cbegin() const { return m_items.cbegin(); }
+  VECFILEITEMS::const_iterator cend() const { return m_items.cend(); }
 private:
   void Sort(FILEITEMLISTCOMPARISONFUNC func);
   void FillSortFields(FILEITEMFILLFUNC func);

--- a/xbmc/cores/playercorefactory/PlayerSelectionRule.cpp
+++ b/xbmc/cores/playercorefactory/PlayerSelectionRule.cpp
@@ -52,7 +52,13 @@ void CPlayerSelectionRule::Initialize(TiXmlElement* pRule)
   m_tBD = GetTristate(pRule->Attribute("bd"));
   m_tDVD = GetTristate(pRule->Attribute("dvd"));
   m_tDVDFile = GetTristate(pRule->Attribute("dvdfile"));
-  m_tDVDImage = GetTristate(pRule->Attribute("dvdimage"));
+  m_tDiscImage = GetTristate(pRule->Attribute("discimage"));
+  if (m_tDiscImage < 0)
+  {
+    m_tDiscImage = GetTristate(pRule->Attribute("dvdimage"));
+    if (m_tDiscImage >= 0)
+      CLog::Log(LOGWARNING, "\"dvdimage\" tag is deprecated. use \"discimage\"");
+  }
 
   m_protocols = XMLUtils::GetAttribute(pRule, "protocols");
   m_fileTypes = XMLUtils::GetAttribute(pRule, "filetypes");
@@ -126,7 +132,7 @@ void CPlayerSelectionRule::GetPlayers(const CFileItem& item, std::vector<std::st
     return;
   if (m_tDVDFile >= 0 && (m_tDVDFile > 0) != item.IsDVDFile())
     return;
-  if (m_tDVDImage >= 0 && (m_tDVDImage > 0) != item.IsDiscImage())
+  if (m_tDiscImage >= 0 && (m_tDiscImage > 0) != item.IsDiscImage())
     return;
 
   CRegExp regExp(false, CRegExp::autoUtf8);

--- a/xbmc/cores/playercorefactory/PlayerSelectionRule.h
+++ b/xbmc/cores/playercorefactory/PlayerSelectionRule.h
@@ -42,7 +42,7 @@ private:
   int m_tBD;
   int m_tDVD;
   int m_tDVDFile;
-  int m_tDVDImage;
+  int m_tDiscImage;
 
   std::string m_protocols;
   std::string m_fileTypes;

--- a/xbmc/games/windows/GUIWindowGames.cpp
+++ b/xbmc/games/windows/GUIWindowGames.cpp
@@ -265,6 +265,13 @@ bool CGUIWindowGames::GetDirectory(const std::string &strDirectory, CFileItemLis
   if (!content.empty())
     items.SetContent(content);
 
+  // Ensure a game info tag is created so that files are recognized as games
+  for (const CFileItemPtr& item : items)
+  {
+    if (!item->m_bIsFolder)
+      item->GetGameInfoTag();
+  }
+
   return true;
 }
 

